factory/factory.go

```diff
@@ -133,7 +133,7 @@ func (fa *Factory) init() {
 		vf := rv.Field(i)
 		ag := &attrGenerator{}
 
-		if tf.Type.Kind() == reflect.Ptr && vf.IsNil() {
+		if (tf.Type.Kind() == reflect.Ptr && vf.IsNil()) || !isExported(tf) {
```

Comment:
@cforbes 
It seems that your `isExpored` has a bug in case that a field has prefix  '_' or '日本' such multibytes.
Could you use `if !vf.CanSet() || (tf.Type.Kind() == reflect.Ptr && vf.IsNil())` instead of this statement?